packages/components/src/icon-button/style.scss

```diff
@@ -22,17 +11,18 @@
 	}
 
 	&.has-text svg {
-		margin-right: 4px;
+		margin-right: 8px;
 	}
 
-	&:not(:disabled):not([aria-disabled="true"]):not(.is-default):hover {
+	// Ideally, we should remove these styles, but they're kept
+	// Because historically, the icon buttons have different hover styles
+	// than the regular buttons.
```

Comment:
Right now we expect icon buttons across Gutenberg UI to have the dark borders on hover while the default buttons (without any variations) don't have this UI. This forces us to have these overrides here.

Ideally, we don't need this and the IconButton should only have to deal with icons, everything else should be deferred to the regular Button component and style. 

cc @jasmussen

Reply:
This is fixed in the last commit 5d35623 but it means that if use a Button without specifying a variation and we expect a custom hover styles, we'd have to override the default, `!important` is the simplest way to achieve that.